symphony/bdk/core/datafeed/abstract_datafeed_loop.py

```diff
@@ -0,0 +1,93 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+from symphony.bdk.core.auth.auth_session import AuthSession
+
+from symphony.bdk.gen.agent_api.datafeed_api import DatafeedApi
+from symphony.bdk.core.datafeed.real_time_event_listener import RealTimeEventListener
+from symphony.bdk.gen.agent_model.v4_event import V4Event
+
+
+class DatafeedLoop(ABC):
+    """Interface for a loop service to be used for handling the datafeed API.
+    """
+
+    @abstractmethod
+    def start(self):
+        """Start the datafeed event service"""
+        pass
+
+    @abstractmethod
+    def stop(self):
+        """Stop the datafeed event service"""
+        pass
+
+    @abstractmethod
+    def subscribe(self, listener: RealTimeEventListener):
+        """The bot subscribes to a RealTimeEventListener
+
+        :param listener: RealTimeEventListener a Datafeed event listener to be subscribed
+        """
+        pass
+
+    @abstractmethod
+    def unsubscribe(self, listener: RealTimeEventListener):
+        """The bot unsubscribes to a RealtimeEventListener
+
+        :param listener: RealTimeEventListener a Datafeed event listener to be subscribed
+        """
+        pass
+
+
+class AbstractDatafeedLoop(DatafeedLoop, ABC):
+    def __init__(self, datafeed_api: DatafeedApi, auth_session: AuthSession, config: BdkConfig):
+        self.datafeed_api = datafeed_api
+        self.listeners = []
+        self.auth_session = auth_session
+        self.bdk_config = config
+        self.api_client = datafeed_api.api_client
+        self.dispatch_dict = {
+            'MESSAGESENT': ('on_message_sent', 'message_sent'),
```

Comment:
is there a way in Python we could use method reference directly instead of strings?

Reply:
Something like https://stackoverflow.com/questions/1351669/referencing-class-methods-in-class-lists-in-python maybe?

Reply:
I spent some time on that one, I agree with you it could have been better.

The problem with this is that we want to access the user implementation of the `on_message_sent`function. 
We can do it using `listener.on_message_sent`. But we can't really pass the listener to a dict or an enum even.

A better solution was using decorators (@singledispatch) but I found it a bit complex to implement. So I'll implement it as an improvement, once I figure out how it works.